client/src/pages/EditProfile.js

```diff
@@ -0,0 +1,178 @@
+import React from 'react';
+import { Typography, withStyles, Button, TextField } from '@material-ui/core';
+import validator from 'validator';
+
+import { OutlinedSelect } from '../components/Inputs';
+import UploadImages from '../components/UploadImages';
+import FormValidator from '../helpers/form-validation';
+import { editUser } from '../api/users';
+import { locations } from '../dummyData/dropDownItems'
+import { withPageContext } from '../components/pageContext';
+
+const styles = {
+  pageContent: {
+    display: 'flex',
+    justifyContent: 'center'
+  },
+  editProfilePage: {
+    margin: '10px',
+    padding: '20px',
+  },
+  form: {
+    display: 'flex',
+    flexDirection: 'column',
+    alignItems: 'left',
+    width: '100%',
+    marginTop: '30px'
+  },
+  formLine: {
+    marginBottom: '20',
+  },
+  button: {
+    margin: '20px 0px 10px 0px'
+  }
+}
+
+class EditProfile extends React.Component {
+  constructor(props) {
+    super(props);
+    this.validators = FormValidator([
+      {
+        field: 'name',
+        method: validator.isEmpty,
+        validWhen: false,
+        message: 'Name is Required'
+      },
+    ])
+    this.state = {
+      name: '',
+      about: '',
+      image: [],
+      location: '',
+      formData: {},
+      validation: this.validators.valid()
+    }
+  }
+
+  componentDidMount() {
+    const { name, about, profilePic, location } = this.props.location.state;
+    this.setState({ name, about, image: [profilePic], location, formData: new FormData() });
+  }
+
+  handleInput = (event) => {
+    const { value, name } = event.target;
+    this.setState({ [name]: value });
+  }
+
+  handleSubmit = async (event) => {
+    event.preventDefault();
+
+    const { name, about, image, location, formData } = this.state;
+    const validation = this.validators.validate(this.state);
+    this.setState({ validation });
+
+    if (validation.isValid) {
+      formData.set('name', name);
+      formData.set('about', about);
+      formData.set('location', location);
+      if (typeof image[0] === 'string') {
+        formData.set('profilePic', image[0])
+      } else {
+        formData.set('image', image[0]);
+      }
```

Comment:
why typeof conditional? They don't seem different

Reply:
This is because when it is saved in the db, it is a string of the aws s3 link, but when you're uploading an image, its a File Object. So I made it so it will not upload an image to AWS in the backend if it's just a string (also it will not upload to aws unless it's a File Object so it works out)